homeassistant/components/evohome/__init__.py

```diff
@@ -81,55 +81,58 @@ def _handle_exception(err) -> bool:
     try:
         raise err
 
-    except evohomeclient2.AuthenticationError:
+    except evohomeasync2.AuthenticationError:
         _LOGGER.error(
             "Failed to (re)authenticate with the vendor's server. "
+            "Check your network and the vendor's service status page. "
             "Check that your username and password are correct. "
             "Message is: %s",
             err,
         )
         return False
 
-    except requests.exceptions.ConnectionError:
+    except aiohttp.ClientConnectionError:
         # this appears to be common with Honeywell's servers
         _LOGGER.warning(
             "Unable to connect with the vendor's server. "
-            "Check your network and the vendor's status page."
+            "Check your network and the vendor's service status page. "
             "Message is: %s",
             err,
         )
         return False
 
-    except requests.exceptions.HTTPError:
-        if err.response.status_code == HTTP_SERVICE_UNAVAILABLE:
+    except aiohttp.ClientResponseError:
+        if err.status == HTTP_SERVICE_UNAVAILABLE:
             _LOGGER.warning(
-                "Vendor says their server is currently unavailable. "
-                "Check the vendor's status page."
+                "The vendor says their server is currently unavailable. "
+                "Check the vendor's service status page."
             )
             return False
 
-        if err.response.status_code == HTTP_TOO_MANY_REQUESTS:
+        if err.status == HTTP_TOO_MANY_REQUESTS:
             _LOGGER.warning(
                 "The vendor's API rate limit has been exceeded. "
-                "Consider increasing the %s.",
+                "If this message persists, consider increasing the %s.",
                 CONF_SCAN_INTERVAL,
             )
             return False
 
-        raise  # we don't expect/handle any other HTTPErrors
+        raise  # we don't expect/handle any other ClientResponseError
 
 
-def setup(hass: HomeAssistantType, hass_config: ConfigType) -> bool:
+async def async_setup(hass: HomeAssistantType, hass_config: ConfigType) -> bool:
     """Create a (EMEA/EU-based) Honeywell evohome system."""
     broker = EvoBroker(hass, hass_config[DOMAIN])
-    if not broker.init_client():
+    if not await broker.init_client():
         return False
 
-    load_platform(hass, "climate", DOMAIN, {}, hass_config)
+    await async_load_platform(hass, "climate", DOMAIN, {}, hass_config)
```

Comment:
Don't await this. Create a task that loads the platform to avoid a deadlock. Use `hass.async_create_task`.

Reply:
Does HA have any opinion re: 
```python
from ... import async_create_task
    ...
    async_create_task()
```
vs
```python
    hass.async_create_task(...)
```
???

Reply:
The former doesn't work. `async_create_task` is a method on the `hass` instance. Please use the latter.